crates/server/src/datastore/mod.rs

```diff
@@ -259,4 +259,10 @@ pub trait DataStore: Send + Sync {
         log_id: &LogId,
         record: &ProtoEnvelope<package::PackageRecord>,
     ) -> Result<(), DataStoreError>;
+
+    // Returns a list of package names, for debugging only.
+    #[doc(hidden)]
+    async fn debug_list_package_ids(&self) -> anyhow::Result<Vec<PackageId>> {
+        anyhow::bail!("not implemented")
+    }
 }
```

Comment:
Nit: Can we simply call this `get_package_ids` rather than have a `debug_` prefix? It feels like the fact that this is only called from the debug endpoint is leaking into the data store API a little.

Also, could this return `DataStoreError`? If this ever ends up being called from a "non-debug" endpoint, it would be nice for callers to be able to differentiate between an error like the postgres database is down (and return a 500 with no context) vs. something else.

Reply:
I'd be happy to take a different approach, but the issue is that this method just operationally doesn't make sense for production; even if there is some production use case for getting all package IDs, you'd probably want this to return a `Stream`, which is unnecessarily complicated to implement for a debug interface.

Reply:
That's a good point re: streaming.

Should it be `cfg(feature = "debug")` to ensure it isn't normally present on the trait (and impls) given its inefficiencies, then?

Reply:
Ah yes, that does make sense.